src/Orc.Controls/Orc.Controls.Shared/Controls/DateTimePicker/Views/DateTimePickerControl.xaml.cs

```diff
@@ -245,6 +294,209 @@ public override void OnApplyTemplate()
             base.OnApplyTemplate();
 
             AccentColorBrush = TryFindResource("AccentColorBrush") as SolidColorBrush;
+
+            EnableOrDisableHourConverterDependingOnFormat();
+        }
+
+        private void OnFormatChanged()
+        {
+            ApplyFormat();
+        }
+
+        private void ApplyFormat()
```

Comment:
Can we unit test this logic somehow? Would love to see some test cases so we are sure this works correctly.